packages/fs-detectors/src/detectors/filesystem.ts

```diff
@@ -96,12 +96,13 @@ export abstract class DetectorFilesystem {
   /**
    * Writes a file to the filesystem cache.
    * @param name the name of the file to write
-   * @param content The content of the file
+   * @param exists true if the files exists, false otherwise
+   * @param content contents of the file
    */
-  public writeFile(name: string, content?: string): void {
-    if (content)
+  public writeFile(name: string, exists?: boolean, content?: string): void {
```

Comment:
In what scenario would you call `writeFile('somename')` without any content?

Reply:
In my PR description I tried to be explicit on usage.  In this case, we can prime the `hasPath` and `isFile` caches like this (think framework detection)

```ts
// we can read the contents of the root directory of a git repo via the `DetectorFilesystem`
const files = fs.readdir('./');
for (const file of files) {
  if (file.type === 'file') {
    // we know this file exists, mark it as so on the filesystem
    // this primes the `hasPath` and `isFile` caches
    fs.writeFile(file.name, undefined, { exists: true });
  }
}
```

Reply:
Maybe it's better to mark `content` as a required parameter here 🤔

Reply:
@styfle Made `content` a required parameter again - thank you for the suggestions!  Feel free to mark this convo as resolved if you are satisfied.

Reply:
I understand this function is not async, however since it shares the same name as Node's `fs.writeFile()`, it might confuse people thinking this is Node's writeFile or that this function is async. What if someday we want to add something async, but this function name is already used for synchronous logic. I'd consider renaming it to something else like `addFile()` or appending "Sync" to the name.

Reply:
@cb1kenobi Its the same as `import fs from 'fs/promises`. I think the name is fine.